nomos-blend/core/src/conn_maintenance.rs

```diff
@@ -349,12 +352,57 @@ mod tests {
 
         let (peers_to_close, peers_to_connect) = maintenance.reset();
         // Peer 0 is malicious
-        assert_eq!(peers_to_close, HashSet::from_iter(vec![peers[0].clone()]));
+        let peer_0_set = HashSet::from_iter(vec![peers[0].clone()]);
+        assert_eq!(peers_to_close, peer_0_set);
+        assert_eq!(maintenance.malicious_peers, peer_0_set);
         // Because Peer 1 is malicious and Peer 2 is unhealthy, 2 new connections should be established
         assert_eq!(peers_to_connect.len(), 2);
```

Comment:
Could you add an explanation to this asserts? I do not quite get them.

Reply:
3a558d04de5f6ecde3d19f106281055ee8510388 I added a comment. You may feel that the returned information is little duplicate. We'll probably make this better while implementing connection manager properly.